modules/telco-core-cpu-partitioning-and-performance-tuning.adoc

```diff
@@ -0,0 +1,41 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/telco_core_ref_design_specs/telco-core-rds.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="telco-core-cpu-partitioning-and-performance-tuning_{context}"]
+= CPU partitioning and performance tuning
+
+New in this release::
+* No reference design updates in this release
+
+Description::
+CPU partitioning separates sensitive workloads from general-purpose tasks, interrupts, and driver work queues, improving performance and reducing latency.
+The CPUs allocated to those auxiliary processes are referred to as reserved in the following sections.
+In Hyper-Threaded systems, a CPU is one hyper-thread.
+
+Limits and requirements::
+* The operating system needs a certain amount of CPU to perform all the support tasks including kernel networking.
+** A system with just user plane networking applications (DPDK) needs at least one Core (2 hyper-threads when enabled) reserved for the operating system and the infrastructure components.
+* In a system with Hyper-Threading enabled, core sibling threads must always be in the same pool of CPUs.
+* The set of reserved and isolated cores must include all CPU cores.
+* Core 0 of each NUMA node must be included in the reserved CPU set.
+* Low latency workloads require special configuration to avoid being affected by interrupts, kernel scheduler, or other parts of the platform.
+For more information, see "Creating a performance profile".
+
+Engineering considerations::
+* The minimum reserved capacity (`systemReserved`) required can be found by following the guidance in  link:https://access.redhat.com/solutions/5843241[Which amount of CPU and memory are recommended to reserve for the system in OpenShift 4 nodes?]
+* The actual required reserved CPU capacity depends on the cluster configuration and workload attributes.
+* The reserved CPU value must be rounded up to a full core (2 hyper-threads) alignment.
+* Changes to CPU partitioning causes the nodes contained in the relevant `MachineConfigPool` to be drain and rebooted.
+* The reserved CPUs reduce the pod density, as the reserved CPUs are removed from the allocatable capacity of the OpenShift node.
```

Comment:
Per ISG: [Do not use _as_ to mean because or while](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#a).

```suggestion
* The reserved CPUs reduce the pod density, because the reserved CPUs are removed from the allocatable capacity of the OpenShift node.
```